app/controllers/admin/media_files_controller.rb

```diff
@@ -13,9 +15,31 @@ def create
       end
 
       authorize @parent, :upload?
+
+      # Bypass Heroku's 30s request timeout by treating it as a streaming response:
+      # https://devcenter.heroku.com/articles/request-timeout#long-polling-and-streaming-responses
+      headers['Content-Type'] = 'application/json'
+      # Prevent Rack::ETag from killing the streaming response: https://git.io/Jk5i0
+      headers['Last-Modified'] = '0'
+      response_keep_alive_thread = Thread.new do
+        loop do
+          sleep 1
+          response.stream.write "\n"
+        rescue IOError => e
```

Comment:
It's unclear to me if it's likely we end up here or not. The error would occur if `response.stream` has been closed at the time we try to write to it, which potentially might occur if we haven't managed to kill the thread in time before the response is fully completed. Concurrency is hard to predict 😬